tests/hashing/main.c

```diff
@@ -0,0 +1,373 @@
+                buf[j] = '0';
+            } else {
+                buf[j]++;
+                break;
+            }
+        }
+        /* Special case if a number that was rounded up only contained the digit 9 */
+        if ((!is_negative && buf[0] == '0' && buf[1] == '0')
+            || (is_negative && buf[1] == '0' && buf[2] == '0')) {
+                for (size_t k=len-1; k>0; k--) {
+                    buf[k] = buf[k-1];
+                }
+                if (!is_negative) {
+                    buf[0] = '1';
+                } else {
+                    buf[1] = '1';
+                }
+                buf[len] = '\0';
+                if (buf[len-1] == '.') {
+                    buf[len-1] = '\0';
+                }
+        }
+    }
+}
+
+#ifdef SHOW_PROGRESS
+
+/*
+    This thread prints the current benchmarking progress
+    (i.e. the number of iterations finished)
+    Native board only
+*/
+void *prog_thread(void *arg)
+{
+    uint32_t *i = arg;
+    msg_t msg;
+
+    while (1) {
+        xtimer_sleep(1);
+        msg_try_receive(&msg);
+        if (msg.content.value == 1) {
+            msg.content.value = 0;
+            printf("\n");
+            thread_sleep();
+        } else if (msg.content.value == 2) {
+            printf("\n");
+            return NULL;
+        } else {
+            printf("Progress: %"PRIu32"\r", *i);
+        }
+    }
+}
+
+#endif
+
+int main(void) {
+
+    printf( "\n\nMeasure performance of SHA256 and Keccak in ticks needed to calculate "
+            "%d hash operations and in hash operations per tick.\n\n", num_iterations);
+
+    /* Initialize */
+    int32_t it_counter = 0;
+    xtimer_ticks64_t start_ticks;
+    xtimer_ticks64_t end_ticks;
+    uint64_t ticks_dif;
+    char ticks_buf[32];
+
+    /*
+        This thread prints the current benchmarking progress
+        (i.e. the number of iterations finished)
+        Native board only
+    */
+    kernel_pid_t prog_thread_pid =
+                thread_create(prog_thread_stack, sizeof(prog_thread_stack),
+                                THREAD_PRIORITY_MAIN - 1, THREAD_CREATE_SLEEPING,
+                                prog_thread, &it_counter, "prog_thread");
+    msg_t msg;
+
+    /* Iterate through all desired string lengths */
+    for (uint32_t k=0; k<sizeof(databytelens)/sizeof(databytelens[0]); k++) {
+
+        size_t databytelen = databytelens[k];
+
+        /* Generate random string */
+        printf("\nGenerating random string of length %d.\n\n", databytelen);
+        random_init(0x33799f);
+        char* datastring = malloc(databytelen);
+        if (datastring == NULL) {
+            printf("\nNot enough memory for further benchmarks.\n\n");
+            return 0;
+        }
+        for (uint32_t l=0; l<databytelen; l++) {
+            datastring[l] = random_uint32_range(0x00, 0xff);
+        }
+        BitSequence* data = (unsigned char*) datastring;
+
+        /* Measure performance of SHA256 */
```

Comment:
I would say that this PR should be simplified by only including
- the Keccak package
- basic Keccak unit-tests, 
i.e. no benchmarks, and no comparison with SHA-256.

Reply:
So basically you suggest I remove the entire tests/hashing directory from the PR?

Reply:
Yes.